lib/proxy/helpers.rb

```diff
@@ -30,7 +30,12 @@ def log_halt(code = nil, exception_or_msg = nil, custom_msg = nil)
 
   # read the HTTPS client certificate from the environment and extract its CN
   def https_cert_cn
-    certificate_raw = request.env['SSL_CLIENT_CERT'].to_s
+    if request.env['HTTP_SSL_CLIENT_CERT']
```

Comment:
This can be spoofed by any user. Reverse proxies without the appropriate middleware are **dangerous**. I'd suggest to read up on this, for example https://telekom-security.github.io/2020/05/smuggling-http-headers-through-reverse-proxies.html.

What this patch now does is that for the default deployment the user can send any certificate using `curl -H 'SSL-Client-Cert: <certificate>' https://smart-proxy.example.com:8443`. It doesn't have to be signed by the correct CA and the CN on it will be used.

If you forget to clear the header in the reverse proxy, you open yourself to the same security issue.

In short: ff you're applying a reverse proxy for security, you must **really** know what you're doing because it's trivial to open gaping security problems.